src/useStorageValue/useStorageValue.ts

```diff
@@ -1,336 +1,267 @@
     };
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isolated, key]);
-
-  return [
-    state,
-    useCallback(
-      (newState) => {
-        if (!isBrowser) return;
-
-        const s = resolveHookState(newState, stateRef.current);
-
-        if (methods.current.storeVal(s)) {
-          methods.current.setState(s);
-
-          if (!isolatedRef.current) {
-            // update all other hooks state
-            storageKeysUsed
-              .get(storage)
-              ?.get(keyRef.current)
-              ?.forEach((setter) => {
-                if (setter === methods.current.setState) return;
-
-                setter(s);
-              });
-          }
-        }
-      },
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-      []
-    ),
-    useCallback(() => {
-      if (!isBrowser) return;
+  }, [storage, key]);
 
-      methods.current.removeVal();
-      methods.current.setState(null);
+  const actions = useSyncedRef({
+    set: (val: NextState<Type, UseStorageValueValue<Type, Default, Initialize>>) => {
+      if (!isBrowser) return;
 
-      if (!isolatedRef.current) {
-        // update all other hooks state
-        storageKeysUsed
-          .get(storage)
-          ?.get(keyRef.current)
-          ?.forEach((setter) => {
-            if (setter === methods.current.setState) return;
+      const s = resolveHookState(
+        val,
+        stateRef.current as UseStorageValueValue<Type, Default, Initialize>
+      );
 
-            setter(null);
-          });
+      const storeVal = storageActions.current.store(s);
+      if (storeVal !== null) {
+        invokeStorageKeyListeners(storage, key, storeVal);
       }
+    },
+    delete: () => {
+      if (!isBrowser) return;
 
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []),
-    useCallback(() => {
+      storageActions.current.remove();
+      invokeStorageKeyListeners(storage, key, null);
+    },
+    fetch: () => {
       if (!isBrowser) return;
 
-      const newVal = methods.current.fetchState();
-      if (newVal !== null && !isolatedRef.current) {
-        // update all other hooks state
-        storageKeysUsed
-          .get(storage)
-          ?.get(keyRef.current)
-          ?.forEach((setter) => {
-            if (setter === methods.current.setState) return;
-
-            setter(newVal);
-          });
-      }
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []),
-  ];
-}
+      invokeStorageKeyListeners(storage, key, storageActions.current.fetchRaw());
+    },
+  });
 
-const storageKeysUsed = new Map<Storage, Map<string, Set<CallableFunction>>>();
+  return useMemo(
+    () => ({
+      value: state as UseStorageValueValue<Type, Default, Initialize>,
+      set: actions.current.set,
+      remove: actions.current.delete,
+      fetch: actions.current.fetch,
+    }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [state]
```

Comment:
The same thing here with the lint rule. This dependency array could also include `actions`, because it too is a ref.

(I always pay attention to disabling of this rule, because I've seen it abused too many times at work, causing mysterious and hard-to-debug behavior. Not saying that's the case here.)

```suggestion
    [state, actions]

Reply:
As for here it is simply useless to add it to dependencies as actions never changes, therefore it would be useless comparison on every render.

Reply:
Yeah, it does not matter - personally I woud rather add it than add the comment, but does not really matter